### PR TITLE
Task/admin page - Implemented Admin Page

### PR DESCRIPTION
**Pull request by N0ot-No0t**
# Description

Added an admin page where the admin user can add accounts and modify the other account types.

Issue reference: #11

## Branch and Commit name must follow this template
```
Branch template: {ISSUE TYPE}-{Issue number}/{desciption_of_branch}
Commit template: {ISSUE TYPE}-{Issue number}: {description of changes}
```

## Type of change

Please delete options that are not relevant.

- [ ] BUG (Fixing or documenting a bug)
- [x] FEAT (Add a new feature to the project)
- [ ] CHORE (Something that needs to be done outside of the issue)
- [ ] DOCS (Implement documentation)
- [x] TASK (A task that must be done to complete an issue)

# Checklist:

- [x] My pull request references the issue in the comments
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/pages/admin/Admin.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+    const request = await fetch('http://localhost:5000/user/' + user.userID, {
+      method: 'PUT',
+      headers: {
+        Authorization: `bearer ${localStorage.token}`,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ userID: user.userID, name: user.name, role: type, email: user.email })
+    });
+    //const response = await request.json();
```
Comment (Talos10) [inline frontend/src/pages/admin/Admin.tsx]: Dead code to be removed?

Comment (tjosepo) [reply frontend/src/pages/admin/Admin.tsx]: We'll mark it as a bug to be fixed next sprint.

Comment (N0ot-No0t) [reply frontend/src/pages/admin/Admin.tsx]: oopsies

**Review by tjosepo — APPROVED** [review]
lgtm

File: frontend/src/components/container/sidenav/Sidenav.tsx [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
+import jwtDecode from 'jwt-decode';
 
 interface Props {
   showSidenav: boolean;
   toggleSidenav: Dispatch<boolean>;
 }
+let userTokenDecoded: any = null;
+
+if (localStorage.getItem('token') != null || localStorage.getItem('token') != undefined) {
+  userTokenDecoded = jwtDecode(localStorage.getItem('token') as string);
```
Comment (sebastien-blain) [inline frontend/src/components/container/sidenav/Sidenav.tsx:15]: Why do you need to decode the token?

**Review by sebastien-blain — APPROVED** [review]
Cool

File: frontend/src/components/container/sidenav/Sidenav.tsx (hunk 2 above)
Comment (N0ot-No0t) [reply frontend/src/components/container/sidenav/Sidenav.tsx:15]: To get the user role to conditionally render the admin tab in the side bar

Comment (sebastien-blain) [reply frontend/src/components/container/sidenav/Sidenav.tsx:15]: Ah ok

Comment (tjosepo) [reply frontend/src/components/container/sidenav/Sidenav.tsx:15]: To get the user role.

Comment (tjosepo) [reply frontend/src/components/container/sidenav/Sidenav.tsx:15]: Only an Admin user should have access to the Admin page.

**Review by Talos10 — APPROVED** [review]
Great work! For the next sprint, you can implement a more permanent fix for the sidebar rendering and also add an error message which informs you when an account already exists with the email that you are trying to create an account with.

**Pull request merged**